### PR TITLE
Leaflet & OpenLayers WMTS Layer

### PR DESCRIPTION
**Pull request by sp13adt**

### CODE REVIEW

File: web/client/components/map/leaflet/plugins/WMTSLayer.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+const Layers = require('../../../../utils/leaflet/Layers');
+const CoordinatesUtils = require('../../../../utils/CoordinatesUtils');
+const WMSUtils = require('../../../../utils/leaflet/WMSUtils');
+const L = require('leaflet');
+const objectAssign = require('object-assign');
+const {isArray, isEqual} = require('lodash');
+const SecurityUtils = require('../../../../utils/SecurityUtils');
+
+
+L.TileLayer.WMTS = L.TileLayer.extend({
```
Comment (mbarto) [inline web/client/components/map/leaflet/plugins/WMTSLayer.js]: Can we put this one into an external file?

File: web/client/components/map/leaflet/plugins/WMTSLayer.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        }
+        return e
+    }
+});
+
+L.tileLayer.wmts = function(urls, options) {
+    return new L.TileLayer.WMTS(urls, options);
+};
+
+function wmsToLeafletOptions(options) {
```
Comment (mbarto) [inline web/client/components/map/leaflet/plugins/WMTSLayer.js]: wmtsToLeafletOptions

File: web/client/components/map/leaflet/plugins/WMTSLayer.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    var opacity = options.opacity !== undefined ? options.opacity : 1;
+    // NOTE: can we use opacity to manage visibility?
+    return objectAssign({}, options.baseParams, {
+        service: "WMTS",
```
Comment (mbarto) [inline web/client/components/map/leaflet/plugins/WMTSLayer.js]: Are you sure you need this?

File: web/client/components/map/leaflet/plugins/WMTSLayer.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        request: "GetTile",
```
Comment (mbarto) [inline web/client/components/map/leaflet/plugins/WMTSLayer.js]: Are you sure you need this?

File: web/client/components/map/leaflet/plugins/WMTSLayer.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+        const url = L.Util.template(this._urls[this._urlsIndex], {s: this._getSubdomain(tilePoint)});
+
+        return url + L.Util.getParamString(this.wmtsParams, url, true) + "&tilematrix=" + ident + "&tilerow=" + tilerow + "&tilecol=" + tilecol
+    },
+        getDefaultMatrix: function() {
+        var e = new Array(22);
+        for (var t = 0; t < 22; t++) {
+            e[t] = {
+                identifier: "EPSG:900913:" + t,
```
Comment (mbarto) [inline web/client/components/map/leaflet/plugins/WMTSLayer.js]: You should use the tileMatrixPrefix option here, not a fixed one

File: web/client/components/map/leaflet/plugins/WMTSLayer.js [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+                topLeftCorner: new L.LatLng(20037508.3428, -20037508.3428)
```
Comment (mbarto) [inline web/client/components/map/leaflet/plugins/WMTSLayer.js]: This is not fixed either, you should allow for configurability (through options) and default to origin of the map crs

File: web/client/components/map/leaflet/plugins/WMTSLayer.js [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+function wmsToLeafletOptions(options) {
+    var opacity = options.opacity !== undefined ? options.opacity : 1;
+    // NOTE: can we use opacity to manage visibility?
+    return objectAssign({}, options.baseParams, {
+        service: "WMTS",
+        request: "GetTile",
+        layer: options.name,
+        styles: options.style || "",
+        format: options.format || 'image/png',
+        tileMatrixSet: "EPSG:900913",
```
Comment (mbarto) [inline web/client/components/map/leaflet/plugins/WMTSLayer.js]: this should be an options param:
options.tileMatrixSet ||  CoordinatesUtils.normalizeSRS(options.srs || 'EPSG:3857', options.allowedSRS)

File: web/client/components/map/leaflet/plugins/WMTSLayer.js (hunk 1 above)
Comment (sp13adt) [inline web/client/components/map/leaflet/plugins/WMTSLayer.js]: I dont really understand what needs to be done in this part.
Do you mean to load all the const from an external file? Like import them?

Comment (mbarto) [inline web/client/components/map/leaflet/plugins/WMTSLayer.js]: No, just move the L.TileLayer.WMTS definition to an external file and require it from here

File: web/client/components/map/leaflet/plugins/index.js [hunk 8]
```diff
@@ -1,19 +1,19 @@
-/**
```
Comment (mbarto) [inline web/client/components/map/leaflet/plugins/index.js]: Do not reformat code (probably editor settings for newlines or tabs/spaces). We have and .editorconfig setting that should be usable by your editor (maybe with some plugin)

File: web/client/components/map/leaflet/plugins/WMTSLayer.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+
+L.tileLayer.wmts = function(urls, options) {
+    return new WMTS(urls, options);
+};
+
+function wmtsToLeafletOptions(options) {
+    // NOTE: can we use opacity to manage visibility?
+    return objectAssign( {}, options.baseParams, {
+        service: "WMTS",
```
Comment (mbarto) [inline web/client/components/map/leaflet/plugins/WMTSLayer.js]: We don't need this, handled by the leaflet plugin

File: web/client/components/map/leaflet/plugins/WMTSLayer.js [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+        request: "GetTile",
```
Comment (mbarto) [inline web/client/components/map/leaflet/plugins/WMTSLayer.js]: We don't need this, handled by the leaflet plugin

**Comment by MV88** [comment]
I think we can close this

**Comment by mbarto** [comment]
Replace by #1340 and #1341

**Pull request closed without merging**